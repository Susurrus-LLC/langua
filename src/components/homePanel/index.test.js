import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import HomePanel from '.'

describe('HomePanel', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <HomePanel linkTo='test' toolTitle='Test'>
        Test
      </HomePanel>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately with various props', () => {
    let tree = renderer
      .create(
        <HomePanel linkTo='test' toolTitle='Test'>
          Test
        </HomePanel>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer
      .create(
        <HomePanel linkTo='different/test' toolTitle='Different Test' disabled>
          Different Test
        </HomePanel>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
