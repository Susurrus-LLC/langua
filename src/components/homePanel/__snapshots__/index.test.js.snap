// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePanel renders accurately with various props 1`] = `
<a
  className="panel"
  href="test"
>
  <h2
    className="toolTitle"
  >
    Test
  </h2>
  <p
    className="toolDescription"
  >
    Test
  </p>
</a>
`;

exports[`HomePanel renders accurately with various props 2`] = `
<a
  className="panel disabled"
  href="different/test"
>
  <h2
    className="toolTitle"
  >
    Different Test
  </h2>
  <p
    className="toolDescription"
  >
    Different Test
  </p>
</a>
`;
